connectors/jdbc/translator-jdbc/src/main/java/org/teiid/translator/jdbc/druid/DruidExecutionFactory.java

```diff
@@ -0,0 +1,531 @@
+            throws TranslatorException {
+        return new DruidQueryExecution(command, conn, executionContext, this);
+    }
+
+    @Override
+    public List<?> translateCommand(Command command, ExecutionContext context) {
+        /*
+        if (command.getClass().toString().equals("class org.teiid.language.Select")) {
+            if (null!= ((Select) command).getGroupBy() && ((Select) command).getGroupBy().getElements().size() > 0) {
+                // remove any columns from group by that are not in the select
+                ((Select) command).getGroupBy().getElements().removeAll(
+                        ((Select) command).getGroupBy().getElements().stream().filter(
+                                g -> !((Select) command).getDerivedColumns().stream().anyMatch(
+                                        d -> d.toString().equals(g.toString()))).collect(Collectors.toList()));
+            }
+        }*/
+        return super.translateCommand(command, context);
+    }
+
+        /*
+    @Override
+    public String translateLiteralBoolean(Boolean booleanValue) {
+        if(booleanValue.booleanValue()) {
+            return "TRUE"; //$NON-NLS-1$
+        }
+        return "FALSE"; //$NON-NLS-1$
+    }
+    */
+
+    @Override
+    public String translateLiteralDate(java.sql.Date dateValue) {
+        return "DATE '" + formatDateValue(dateValue) + "'"; //$NON-NLS-1$//$NON-NLS-2$
+    }
+
+    @Override
+    public String translateLiteralTime(Time timeValue) {
+        return "TIME '" + formatDateValue(timeValue) + "'"; //$NON-NLS-1$//$NON-NLS-2$
+    }
+
+    @Override
+    public String translateLiteralTimestamp(Timestamp timestampValue) {
+        if (timestampValue.getNanos() == 0) {
+            String val = formatDateValue(timestampValue);
+            val = val.substring(0, val.length() - 2);
+            return "TIME_PARSE('" + val + "', '" + DATETIME_FORMAT + "')"; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+        }
+        return super.translateLiteralTimestamp(timestampValue);
+    }
+
+    @Override
+    public boolean usePreparedStatements() {
+        return false;
+    }
+
+    @Override
+    public int getTimestampNanoPrecision() {
+        return 6;
+    }
+
+    /**
+     * Postgres doesn't provide min/max(boolean), so this conversion writes a min(BooleanValue) as
+     * bool_and(BooleanValue)
+     * @see org.teiid.language.visitor.LanguageObjectVisitor#visit(org.teiid.language.AggregateFunction)
+     * @since 4.3
+     */
+    @Override
+    public List<?> translate(LanguageObject obj, ExecutionContext context) {
+        if (obj.getClass().toString().equals("class org.teiid.language.Select")) {
+            if (null!= ((Select) obj).getGroupBy() && ((Select) obj).getGroupBy().getElements().size() > 0) {
+                // remove any columns from group by that are not in the select
+                ((Select) obj).getGroupBy().getElements().removeAll(
```

Comment:
You probably don't want to do this here - the projected columns won't line up and you should get exceptions later on.  You probably need a capability similar to supportsOrderByUnrelated - where the engine won't allow for grouping expressions that don't appear in the select.